interpreter/valid/valid.ml

```diff
@@ -183,7 +183,9 @@ let check_func_type (c : context) (ft : func_type) at =
 
 let check_cont_type (c : context) (ct : cont_type) at =
   match ct with
-  | ContT ft -> check_heap_type c ft at
+  | ContT (VarHT (StatX x)) ->
+     let _dt = func_type c (x @@ at) in ()
+  | _ -> error at "ill-formed continuation type"
```

Comment:
```suggestion
  | _ -> assert false
```
In source, this cannot be anything else by construction. Analogous to how `sub` handles it.